ipn/store/kubestore/store_kube.go

```diff
@@ -203,6 +204,23 @@ func (s *Store) ReadTLSCertAndKey(domain string) (cert, key []byte, err error) {
 			// that wraps ipn.ErrStateNotExist here.
 			return nil, nil, ipn.ErrStateNotExist
 		}
+		st, ok := err.(*kubeapi.Status)
```

Comment:
This looks like a good place to use `errors.As`, in case we wrap the errors in future.

Reply:
I added a test to the kube client library that should catch this instead, as I find it too easy to mishandle `errors.As`; it requires both the error sender and the error receiver to coordinate perfectly on both the concrete type and whether it implements the error as a struct or as a pointer to a struct.